test/validation/api/ipsec/ipsec.c

```diff
@@ -1155,6 +1163,52 @@ int ipsec_config(odp_instance_t ODP_UNUSED inst)
 	ipsec_config.inbound.chksums.all_chksum = ~0;
 	ipsec_config.stats_en = true;
 
+	ipsec_config.inbound.reassembly.max_wait_time = 20 * ODP_TIME_MSEC_IN_NS;
+	ipsec_config.inbound.reassembly.max_num_frags = capa.reassembly.max_num_frags;
+
+	if (capa.reassembly.ip) {
+		ipsec_config.inbound.reassembly.en_ipv4 = true;
+		ipsec_config.inbound.reassembly.en_ipv6 = true;
+	}
+
+	if (capa.reassembly.ipv4)
+		ipsec_config.inbound.reassembly.en_ipv4 = true;
+
+	if (capa.reassembly.ipv6)
+		ipsec_config.inbound.reassembly.en_ipv6 = true;
+
+	if (ODP_IPSEC_OP_MODE_INLINE == suite_context.inbound_op_mode &&
+	    !capa.reass_inline_mode) {
+		ipsec_config.inbound.reassembly.en_ipv4 = false;
+		ipsec_config.inbound.reassembly.en_ipv6 = false;
+	}
+
+	if (ODP_IPSEC_OP_MODE_ASYNC == suite_context.inbound_op_mode &&
+	    !capa.reass_async_mode) {
+		ipsec_config.inbound.reassembly.en_ipv4 = false;
```

Comment:
This leaves reassembly testing enabled in async mode if supported by the implementation, but the current validation test code does not handle reassembly in the async path. So, either reassembly needs to be disabled here for now, or the async event reception path needs to be modified along the lines I described earlier.

Reply:
I had added the changes you had suggested. But there are other reassembly handling also needed in that path. For now, I'll add an ASSERT and add a comment stating that ASYNC is not supported yet.